.github/workflows/android-build.yml

```diff
@@ -40,48 +40,38 @@ jobs:
         with:
           gradle-version: '8.6'
 
-      # Check the NDK that we're using
-      - name: Check Android NDKs
+      # Check the Android SDK and NDK that we're using
+      - name: Check Android SDK and NDK
         run: |
           set -e -x
           uname -m
+          echo "ANDROID_HOME=$ANDROID_HOME"
+          ls -l $ANDROID_HOME
+
+          echo "ANDROID_SDK_ROOT=$ANDROID_SDK_ROOT"
+
           echo "ANDROID_NDK_HOME=$ANDROID_NDK_HOME"
           echo "ANDROID_NDK_LATEST_HOME=$ANDROID_NDK_LATEST_HOME"
           ls -l $ANDROID_HOME/ndk
 
+      - name: Check if emulator is installed and install it if needed
+        run: |
+          if [ -d "${ANDROID_SDK_ROOT}/emulator" ]; then
+              echo "${ANDROID_SDK_ROOT}/emulator exists"
+          else
+              ${ANDROID_SDK_ROOT}/cmdline-tools/latest/bin/sdkmanager --install "emulator"
+          fi
+
       # Needed for linux
       - name: Install jq
         run: |
           sudo apt-get install jq
 
-      - uses: actions/setup-dotnet@v4
```

Comment:
For some unknown reason, `dotnet install` failed with permission denied.

See https://github.com/microsoft/onnxruntime-genai/actions/runs/12385746487/job/34573153027